tests/invariants/fail/type_constraints_env.act

```diff
@@ -0,0 +1,8 @@
+constructor of AllEnvVars
+interface constructor()
+
+creates
+	uint value := CALLVALUE
```

Comment:
This test should pass. Added to `fail` so that this passes for now, and fails when it's fixed.